array_api_compat/__init__.py

```diff
@@ -19,4 +19,18 @@
 """
 __version__ = '1.4.1'
 
-from .common import *
+from .common import (
```

Comment:
Can ruff help keep these two files in sync?

Reply:
This part I was not happy with either, because the import path is two-level: functions are imported from `array_api_compat/common._helpers.py` into `array_api_compat.common` from their they are imported into `array_api_compat`. So if anything gets added to `_helpers.py` it has to be updated in two places. Maybe one keeps the `*` import here and ignores...

Reply:
I guess it is not important that users can do both `from array_api_compat import get_namespace` and `from array_api_compat.common import get_namespace`. So my proposal would be to decide on one instead? However this would be a backwards incompatible change...

Reply:
Alternatively we could just keep the `*` import and ignore.

Reply:
If ruff can't automatically keep `__all__` synced between a module and submodule with explicit imports then using `import *` and just setting `__all__ = common.__all__` is the next best thing. 

This would be a useful feature for some linter. We do this sort of thing in SymPy, for instance (e.g., at https://github.com/sympy/sympy/blob/master/sympy/core/__init__.py and https://github.com/sympy/sympy/blob/master/sympy/__init__.py), and we have to manually keep them in sync. 

OTOH, having `common` as a submodule that people can import from isn't that important, unlike the `torch`, `numpy`, etc. which are important since those are the actual array API namespaces.